herd/ASLSem.ml

```diff
@@ -385,6 +392,19 @@ module Make (C : Config) = struct
     (* Primitives and helpers                                                 *)
     (**************************************************************************)
 
+    let vbool b =  V.Val (Constant.Concrete (ASLScalar.S_Bool b))
+
+    let checkprop m_prop =
+      let* vprop = m_prop in
+      M.assign (vbool true) vprop >>! []
```

Comment:
Is that different from the ASL syntax `assert prop`?

Reply:
I have not checked, but I assume this is different: `checkprop` produces an equation that will be contradictory if vprop is `FALSE`; while `ASSERT` raises an exception that may have less controlled consequences.

Reply:
Maybe we should implement assertions like this then?

Reply:
I am not sure we should. Assertions are here to check for errors. We probably want errors to be flagged as such and not to disappear silently.

Reply:
Okay then.